src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/JsonHubProtocol.cs

```diff
@@ -64,30 +68,24 @@ public bool TryWriteMessage(HubMessage message, IOutput output)
 
         private HubMessage ParseMessage(Stream input, IInvocationBinder binder)
         {
-            var reader = new JsonTextReader(new StreamReader(input));
-
-            // PERF: Could probably use the JsonTextReader directly for better perf and fewer allocations
-            var json = _payloadSerializer.Deserialize<JObject>(reader);
-            if (json == null)
+            using (var reader = new JsonTextReader(new StreamReader(input)))
             {
-                return null;
-            }
+                // PERF: Could probably use the JsonTextReader directly for better perf and fewer allocations
+                var json = _payloadSerializer.Deserialize<JObject>(reader);
+                if (json == null)
+                {
+                    return null;
```

Comment:
Either handle this (e.g here: https://github.com/aspnet/SignalR/pull/390/commits/fa7ead1f51ac45c4cc928d6725e43cedc45d66b9#diff-b8bab5f1c287065aa78fe0918014bb98R162) or maybe just throw here?